lib/mix/test/mix/tasks/escript_test.exs

```diff
@@ -96,6 +96,28 @@ defmodule Mix.Tasks.EscriptTest do
     end
   end
 
+  defmodule EscriptWithPrivs do
```

Comment:
It is fine to leave this as is, but I think these tests could be written without defining modules. We have a way of pushing updated configuration when we do `Mix.Project.push`, which simplify not having to create a bunch of modules.

We should probably refactor this whole test case to use said style, so it could be done later anyway.

Reply:
Do you mean `Mix.ProjectStack.post_config/1`? I can refactor in a separate PR :)

Reply:
Yes!!